GVFS/GVFS.Common/Tracing/EventListener.cs

```diff
@@ -8,36 +8,41 @@ public abstract class EventListener : IDisposable
         private readonly EventLevel maxVerbosity;
         private readonly Keywords keywordFilter;
 
+        private IEventListenerEventSink eventSink;
+
         protected EventListener(EventLevel maxVerbosity, Keywords keywordFilter)
         {
             this.maxVerbosity = maxVerbosity;
             this.keywordFilter = keywordFilter;
         }
 
+        public void RegisterEventSink(IEventListenerEventSink eventSink)

```

Comment:
Can we have the interface come through the constructor instead of a separate method.  `AddEventListener` is the only place it is used where it doesn't have the construction of the listener but that method is only used in tests and could be `private` if we didn't have the tests which I don't think we should remove so I think we need to keep it but can we skip the `RegisterEventSink` there and leave the responsibility to the constructor of the event listener?

Reply:
Moved to passing the sinks in via the constructors.